chatapp/user/consumers.py

```diff
@@ -0,0 +1,50 @@
+import json
+
+from channels.generic.websocket import AsyncWebsocketConsumer
+
+from user.utils import group_name
+
+
+class ChatConsumer(AsyncWebsocketConsumer):
+    async def connect(self):
+        self.room_group_name = f"chat"
+        await self.channel_layer.group_add(self.room_group_name, self.channel_name)
+        await self.accept()
+
+    async def disconnect(self, close_code):
+        await self.channel_layer.group_discard(self.room_group_name, self.channel_name)
+
+    # Receive message from WebSocket
+    async def receive(self, text_data):
+        text_data_json = json.loads(text_data)
+        message = text_data_json["message"]
+        # Send message to room group
+        await self.channel_layer.group_send(
+            self.room_group_name, {"type": "chat.message", "message": message}
+        )
+
+    async def chat_message(self, event):
+        message = event["message"]
+        await self.send(text_data=json.dumps({"message": message}))
+
+class ChatMessage(AsyncWebsocketConsumer):
+    async def connect(self):
+        self.room_group_name = group_name(self.scope['user'].id,self.scope["url_route"]["kwargs"]["id"])
+        await self.channel_layer.group_add(self.room_group_name, self.channel_name)
```

Comment:
if scope['user'].id is 1 and self.scope["url_route"]["kwargs"]["id"] is 2
it merans that sender is 1 and receiver is 2
group name is 1,2

if receiver is sender and sender is receiver in that case ? what will be the happened ??